### ISSUE TITLE
.NET/ASP Core 2.0 Compatibility

### ISSUE THREAD

**Issue by Silthus** [ivaylokenov/MyTested.AspNetCore.Mvc#278]
When will this framework be compatible with ASP.NET Core 2.0?

Currently, when I try to use the library, I get the following error:

```
System.TypeInitializationException : The type initializer for 'MyTested.AspNetCore.Mvc.MyMvc' threw an exception.
---- System.MissingMethodException : Method not found: 'Void Microsoft.AspNetCore.Hosting.Internal.ApplicationLifetime..ctor()'.
   at MyTested.AspNetCore.Mvc.MyMvc.Controller[TController]()
   at HoehenSucht.Frontend.Integration.Tests.api.TourBooksTests.Test_TourBooks_ReturnsEmptyArray() in C:\Program Files\Temp\Sources\HoehenSucht\test\HoehenSucht.Frontend.Integration.Tests\api\TourBooksTests.cs:line 24
----- Inner Stack Trace -----
   at MyTested.AspNetCore.Mvc.Internal.Application.TestApplication.GetInitialServiceCollection()
   at MyTested.AspNetCore.Mvc.Internal.Application.TestApplication.Initialize()
   at MyTested.AspNetCore.Mvc.Internal.Application.TestApplication.TryInitialize()
   at MyTested.AspNetCore.Mvc.MyMvc..cctor()
```

**Comment by HesamKashefi** [comment]
It seems that the owner has abandoned programming completely! :|  @ivaylokenov

**Comment by chernihiv** [comment]
Dear @ivaylokenov, have you got any plans to support this project anymore?

**Comment by buerobuettner** [comment]
Please support ASP.NET Core 2.0 - a lot of devs will use this version now.

**Comment by Calabonga** [comment]
Please support ASP.NET Core 2.1 - this is feature need very much

**Comment by ivaylokenov** [comment]
@Silthus @chernihiv @buerobuettner @Calabonga 

Hi guys,

The library is now ported to .NET Core 2. I will try to explain why it was delayed by so much time.

Back in March 2017 when .NET Core migrated to the new (old) .csproj file format, I was having difficulties with the project as it can be seen here - https://github.com/ivaylokenov/MyTested.AspNetCore.Mvc/issues/270#issuecomment-290385040

Visual Studio was continually crashing, tests did not run correctly, and the overall developing experience was terrible. I tried for 2-3 months to port the library but could not succeed. 

During that time, I started developing severe health issues from staying in front of my computer too much. I was working 12 to 16 hours from Monday to Sunday with not many breaks for 2-3 years straight. I had to stop doing it for an indefinite time. My GitHub activity shows this pretty clearly: https://github.com/ivaylokenov

The truth is - I like this testing library and have much fun developing it, so I was confused and did not know what to tell people about its future maintenance. I did not want to abandon it forever but did not know when I would be able to work actively again on it. I now realize that it was a mistake to be silent all this time and should have put an explanation earlier. The fact that my personal life was very stressful for the last two years is not an excuse.

Back in October 2018, I opened the code and started working actively on bringing back the library to life. It took me 6 months because I added lots of new features like third-party DI support, all possible action results, every single action filter, lots of helper methods for easier testing, more straightforward configuration, performance fixes and more. Version 2.2 is available on NuGet already, and version 2.1 will be published in the next few days. I will then update the tutorial, documentation, the web site, and will start working on .NET Core 3.

I am sorry for the inconvenience caused by this library and me. I truly am. 

As compensation (if that is even possible), all versions before 3.0 do not have any limitations and can be used without any license code.

The good news is that I found a small private investor who is sponsoring the project. I can actively put 20 to 30 hours per week in the development of MyTested.AspNetCore.Mvc. If I manage to increase the usage and find clients, the investor will increase the budget and I will be able to hire another developer to help me. Additionally, I will put a sponsorship program with logos, and hopefully, I will be able to maintain the project steadily in the years to come.

Thank you for reading and understanding. 

Ivaylo
